src/test/ui/non-fmt-panic.rs

```diff
@@ -29,6 +29,12 @@ fn main() {
     fancy_panic::fancy_panic!(S);
     //~^ WARN panic message is not a string literal
 
+    macro_rules! a {
+        () => { 123 };
+    }
+
+    panic!(a!()); //~ WARN panic message is not a string literal
```

Comment:
Can you add a test where `panic!` is neither the innermost nor outermost macro?

```rust
macro_rules! my_assert() {
  ($condition: expr) => assert!($condition, $condition);
}
my_assert!(false);
```

Reply:
That's already tested with `fancy_panic!()` on line 26. (`fancy_panic!()` -> `panic!()` -> `panic_2015!()` -> final code).